### PR TITLE
adding grover pretrain module

### PR DESCRIPTION
**Pull request by gusty1g**
# Pull Request Template

## Description

Adding GroverPretrain module which will be later initialized by GroverModularTorchModel for pretraining GroverModel.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by gusty1g** [comment]
The changes in [6840630](https://github.com/deepchem/deepchem/pull/3334/commits/684063027fd1c116c78e94f14bdde6332dc36d7b) are a small bug fix in representation of batched grover graph. The earlier grover implementation used padding in batched grover graph which we are not doing anymore, hence those lines were removed.

File: deepchem/models/torch_models/grover.py [hunk 1]
```diff
@@ -4,6 +4,114 @@
 from deepchem.models.torch_models.grover_layers import GroverTransEncoder
 
 
+class GroverPretrain(nn.Module):
```
Comment (rbharath) [inline deepchem/models/torch_models/grover.py:7]: Can you add this to `layers.rst`?

**Review by rbharath — APPROVED** [review]
A small comment requesting this class be added to the docs.

LGTM in the interest of speed. Feel free to merge after adding to the docs and making sure CI is clear

File: deepchem/models/torch_models/grover.py (hunk 1 above)
Comment (gusty1g) [reply deepchem/models/torch_models/grover.py:7]: added docs
